sentry/resource_sentry_project.go

```diff
@@ -284,6 +287,20 @@ func resourceSentryProjectUpdate(ctx context.Context, d *schema.ResourceData, me
 		params.AllowedDomains = allowedDomains
 	}
 
+	if _, ok := d.GetOk("remove_default_key"); ok {
```

Comment:
If the user sets `remove_default_key: false`, I believe this will still remove the default key since `ok` in this line refers to whether `remove_default_key` exists as a key in the resource (not whether it has been set to `true`).

The same will occur on line 297 for `remove_default_rule`. 

Nit (we can skip this part if you've got more pressing work): It might be worth testing that `remove_default_key: false`, does not remove the default key and the same for `remove_default_rule`.

Reply:
I've taken a look at the documentation and it seems that `ok` will be `true` if `remove_default_key` the key exists and it has been set to a non-zero value, but yup I can add tests to see if `remove_default_key` and `remove_default_rule` are set to `false`

Reply:
Whoops. I forgot false === zero value in golang. Thanks for this :) . Probably don't need the test.